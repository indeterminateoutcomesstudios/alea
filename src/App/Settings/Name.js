/* @flow */
import React from 'react'
import styled, { withTheme } from 'styled-components'
import Label from 'components/Label'
import Input from 'components/Input'
import editable from 'containers/editable'

type Props = {
  name: string,
  onChange: Function
}

const Editable = editable(withTheme(styled.div`
  position: relative;
  line-height: 3.6rem;
  background: transparent;
  border: none;
  border-radius: 0;
  padding-left: 0;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  color: ${props => props.theme.color};
  margin: 0 0 1.2rem;
  width: 100%;
  &:hover {
    cursor: pointer;
  }
`))

const NameInput = withTheme(Input.extend`
  background: transparent;
  border: none;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  color: ${props => props.theme.color};
  margin: 0 0 1.2rem;
  width: 100%;
  padding: 0;
`)

const Name = (props: Props) => {
  return <div>
    <Label>Display Name:</Label>
    <Editable
      isHovering={true}
      value={props.name}
      input={NameInput}
      onChange={props.onChange}
    />
  </div>
}

export default Name
