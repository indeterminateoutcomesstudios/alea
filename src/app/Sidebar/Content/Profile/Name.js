/* @flow */
import React from 'react'
import Editable from 'app/components/Editable'
import Label from '../Label'
import sidebarStyles from '../style.css'

type Props = {
  name: string,
  onChange: Function
}

const Name = (props: Props) => {
  return <div>
    <Label>Display Name:</Label>
    <Editable
      className={ sidebarStyles.field }
      value={props.name}
      onChange={val => props.onChange(val)}/>
  </div>
}

export default Name
