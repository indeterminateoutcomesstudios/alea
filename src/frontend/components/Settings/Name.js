// @flow
import React from 'react'
import Input from 'frontend/components/Input'

type Props = {
  name: string,
  saveDisplayName: Function
}

type State = {
  name: string
}

export default class Name extends React.Component<Props, State> {
  state = {
    name: this.props.name
  }

  componentWillReceiveProps(prev: Props) {
    // See if there's a new name
    if (this.props.name !== prev.name) {
      this.setState({ name: this.props.name })
    }
  }

  onChange = (e: SyntheticEvent<HTMLInputElement>) => {
    this.setState({ name: e.currentTarget.value })
  }

  onBlur = () => {
    if (this.state.name != this.props.name) {
      this.props.saveDisplayName(this.state.name)
    }
  }

  render() {
    return (
      <Input
        value={this.state.name}
        onChange={this.onChange}
        onBlur={this.onBlur}
      />
    )
  }
}
