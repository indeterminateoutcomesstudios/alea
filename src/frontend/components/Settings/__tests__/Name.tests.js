// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Name from '../Name.js'

describe('Settings Name component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Name name="name" saveDisplayName={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
