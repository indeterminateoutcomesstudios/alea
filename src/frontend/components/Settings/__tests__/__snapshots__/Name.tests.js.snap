// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Name component renders correctly 1`] = `
.c0 {
  padding: 1rem;
  border: 1px solid transparent;
  border-radius: 4px;
  font-family: -apple-system,BlinkMacSystemFont,  "Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans",  "Helvetica Neue",sans-serif;
  font-size: 1.5rem;
  line-height: 1;
}

.c0:focus {
  background: undefined;
  border-color: undefined;
}

<input
  className="c0"
  onBlur={[Function]}
  onChange={[Function]}
  value="name"
/>
`;
